skimage/morphology/gray.py

```diff
@@ -70,93 +71,48 @@ def _shift_footprint(footprint, shift_x, shift_y):
+    """Shifts the footprints, whether it's a single array or a sequence.
 
-    This is a patch for the *weird* footprint inversion in
-    `ndi.grey_morphology` [1]_.
-
-    Parameters
-    ----------
-    footprint : array
-        The input footprint.
-
-    Returns
-    -------
-    inverted : array, same shape and type as `footprint`
-        The footprint, in opposite order.
-
-    Examples
-    --------
-    >>> footprint = np.array([[0, 0, 0], [0, 1, 1], [0, 1, 1]], np.uint8)
-    >>> _invert_footprint(footprint)
-    array([[1, 1, 0],
-           [1, 1, 0],
-           [0, 0, 0]], dtype=uint8)
-
-    References
-    ----------
-    .. [1] https://github.com/scipy/scipy/blob/ec20ababa400e39ac3ffc9148c01ef86d5349332/scipy/ndimage/morphology.py#L1285  # noqa
+    See `_shift_footprint`, which is called for each array in the sequence.
     """
-    inverted = footprint[(slice(None, None, -1),) * footprint.ndim]
-    return inverted
-
+    if shift_x is None and shift_y is None:
+        return footprint
 
-def pad_for_eccentric_footprints(func):
-    """Pad input images for certain morphological operations.
+    warning_msg = ("The parameters `shift_x` and `shift_y` are deprecated "
+                   "since v0.23 and will be removed in v0.26.")
+    warnings.warn(warning_msg, FutureWarning, stacklevel=4)
 
-    Parameters
-    ----------
-    func : callable
-        A morphological function, either opening or closing, that
-        supports eccentric footprints. Its parameters must
-        include at least `image`, `footprint`, and `out`.
-
-    Returns
-    -------
-    func_out : callable
-        The same function, but correctly padding the input image before
-        applying the input function.
-
-    See Also
-    --------
-    opening, closing.
-    """
-    @functools.wraps(func)
-    def func_out(image, footprint, out=None, *args, **kwargs):
-        pad_widths = []
-        padding = False
-        if out is None:
-            out = np.empty_like(image)
-        if _footprint_is_sequence(footprint):
-            # Note: in practice none of our built-in footprint sequences will
-            #       require padding (all are symmetric and have odd sizes)
-            footprint_shape = _shape_from_sequence(footprint)
-        else:
-            footprint_shape = footprint.shape
-        for axis_len in footprint_shape:
-            if axis_len % 2 == 0:
-                axis_pad_width = axis_len - 1
-                padding = True
-            else:
-                axis_pad_width = 0
-            pad_widths.append((axis_pad_width,) * 2)
-        if padding:
-            image = np.pad(image, pad_widths, mode='edge')
-            out_temp = np.empty_like(image)
+    if _footprint_is_sequence(footprint):
+        return tuple((_shift_footprint(fp, shift_x, shift_y), n)
+                     for fp, n in footprint)
+    return _shift_footprint(footprint, shift_x, shift_y)
+
+
+def _handle_mode_and_cval(dtype, mode, cval):
```

Comment:
How about the name `_min_max_to_constant_mode` instead? It's less generic than "handle".